src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunTableManager.cs

```diff
@@ -4,21 +4,46 @@
 
 using System;
 using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
 
 using ILCompiler.DependencyAnalysis;
 using ILCompiler.DependencyAnalysis.ReadyToRun;
 using ILCompiler.DependencyAnalysisFramework;
 
 using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
 
 using Debug = System.Diagnostics.Debug;
 
 namespace ILCompiler
 {
-    public class ReadyToRunTableManager : MetadataManager
+    public struct DefinedTypeInfo
+    {
+        public readonly EcmaModule Module;
```

Comment:
Nit: What the consumer seems to really need is just a `MetadataReader`.

Reply:
Agreed - at least for now. It remains to be seen what happens when we start putting multiple R2R modules in the same executable. I'm starting to think that perhaps it might be easiest to model them along Morgan's "single file" model - no real merging, the executable would just contain a number of separate COR headers, metadata blobs, R2R headers and separate method & other tables, we would just have a "master directory" somewhere that would let us access the "submodules" within the exe (and map them onto MSIL . This still shouldn't prevent optimizations like intra-bubble inlining.

Reply:
Fixed in 4th commit.